subzero-core/src/main/java/info/jerrinot/subzero/SubZero.java

```diff
@@ -23,22 +23,38 @@ private SubZero() {
      * Use SubZero as a global serializer.
      *
      * This method configures Hazelcast to delegate a class serialization to SubZero when the class
-     * has no explicit strategy configured.
+     * has no explicit strategy configured. Uses {@link Serializer} serializer implementation
+     * internally.
      *
      * @param config Hazelcast configuration to inject SubZero into
      * @return Hazelcast configuration.
      */
     public static Config useAsGlobalSerializer(Config config) {
+        return useAsGlobalSerializerInternal(config, Serializer.class);
+    }
+
+    /**
+     * Use SubZero as a global serializer.
+     *
+     * This method configures Hazelcast to delegate a class serialization to SubZero when the class
+     * has no explicit strategy configured.
+     *
+     * @param config Hazelcast configuration to inject SubZero into
+     * @param serializerClazz Class of global serializer implementation to use
+     * @return Hazelcast configuration.
+     */
+    public static <T> Config useAsGlobalSerializer(Config config, Class<? extends AbstractGlobalUserSerializer> serializerClazz) {
```

Comment:
I would not add this method. Methods in the `SubZero` class are for user-convenience. For a quick-start if the default SubZero is good enough for you. If you are implementing own serializer then you do not need the convenience.

I expect majority of users won't/shouldn't need to create custom subclasses. The extra methods here could be confusing for new users.